### ISSUE TITLE
Reminder: Cairo clip does not actually remove data for certain plot types

### ISSUE THREAD

**Issue by asinghvi17** [MakieOrg/Makie.jl#2100]
Consider the following totally minimal example:
```julia
sc = Scene(camera = cam2d!)
scatter!(sc, 1:10, rand(10))
Makie.update_cam!(sc, Rect2f(Point2f(1, 0), Vec2f(5, 1)))

save("hello.svg", sc; pt_per_unit=1)
```

This has only 5 lines of markers, as it should be.

But, using `lines!` instead of `scatter!`, there are 10 points in the line.  We should probably fix this by skipping the drawing for those segments for which both points are outside of the drawing area indicated by the Scene, and where the line does not intersect the scene (this means that (p1[1], p2[2]) and (p2[1], p1[2]) should both be outside the Scene).  

This would be another check in the main drawing loop of draw_atomic for lines, but has the potential to drastically decrease file-sizes and perhaps draw times, especially for plots on which all available data is plotted but only a section is directly observed (e.g., complex dashboards).

This is most likely also an issue for meshes and heatmaps, both of which we should also look at performing similar optimizations for.

**Comment by jkrumbiegel** [comment]
It's a bit more problematic because a marker or line outside of the scene could still be partially visible depending on how wide/thick it is.

**Comment by asinghvi17** [comment]
Good point.  We should then check linewidth, and add some padding based on that.  It seems that Cairo already takes care of markers.

**Comment by ffreyer** [comment]
I fixed something about line clipping in #3958 though it seems like it's still buggy... GLMakie vs CairoMakie:
![image](https://github.com/user-attachments/assets/de6407e5-d4a6-4577-9737-c63b7a087358)

**Comment by ffreyer** [comment]
There have been a bunch more fixes, i.e. #4313, #4601, #4631. I think it's working well enough now that this can be closed.
